tests/model_registry/model_catalog/test_model_catalog_negative.py

```diff
@@ -0,0 +1,174 @@
+- name: {first_catalog_data["name"]}
+  id: {first_catalog_id}
+  type: {first_catalog_data["type"]}
+  properties:
+    yamlCatalogPath: {first_catalog_data["properties"]["yamlCatalogPath"]}
+"""
+
+    return f"""catalogs:
+{catalog_str}
+"""
+
+
+@pytest.mark.skip_must_gather
+class TestDefaultCatalogNegative:
+    """Negative tests for default catalog configuration"""
+
+    @pytest.mark.parametrize(
+        "model_catalog_config_map, modified_sources_yaml",
+        [
+            pytest.param(
+                {"configmap_name": DEFAULT_MODEL_CATALOG_CFG},
+                """
+catalogs:
+  - name: Modified Catalog
+    id: modified_catalog
+    type: yaml
+    properties:
+      yamlCatalogPath: /shared-data/modified-catalog.yaml
+""",
+                id="test_modify_catalog_structure",
+            ),
+        ],
+        indirect=["model_catalog_config_map"],
+    )
+    def test_modify_default_catalog_configmap_reconciles(
+        self: Self, model_catalog_config_map: ConfigMap, modified_sources_yaml: str
+    ):
+        """
+        Test that attempting to modify the default catalog configmap raises an exception.
+        This validates that the default catalog configmap is protected from modifications.
+        """
+        # Attempt to modify the configmap - this should raise an exception
+        patches = {"data": {"sources.yaml": modified_sources_yaml}}
+
+        with ResourceEditor(patches={model_catalog_config_map: patches}):
+            # This block should raise an exception due to configmap protection
+            LOGGER.info("Attempting to modify protected configmap")
+
+        # Verify the configmap was not actually modified
+        validate_model_catalog_configmap_data(
+            configmap=model_catalog_config_map, num_catalogs=len(DEFAULT_CATALOGS.keys())
+        )
+
+
+@pytest.mark.skip_must_gather
+class TestCustomCatalogNegative:
+    """Negative tests for custom catalog configuration"""
+
+    @pytest.mark.parametrize(
+        "updated_catalog_cm_negative",
+        [
+            pytest.param(
+                {"data": {"sources.yaml": _get_default_catalog_str()}},
+                id="test_default_catalog_in_custom_configmap",
+            ),
+        ],
+        indirect=True,
+    )
+    def test_default_catalog_rejected_in_custom_configmap(
+        self: Self, updated_catalog_cm_negative: ConfigMap, model_catalog_pod_error_state: Pod
+    ):
+        """
+        Test that attempting to add a default catalog to custom configmap is rejected.
+        This validates that default catalogs cannot be added to custom catalog configmap.
+        """
+        # Check model catalog pod logs for the expected error
+        pod_log = model_catalog_pod_error_state.log(container=CATALOG_CONTAINER)
+
+        # Look for error pattern using regex
+        error_pattern = r"Error: error loading catalog sources: (.*): source (.*) exists from multiple origins"
+        match = re.search(error_pattern, pod_log)
```

Comment:
this is quite brittle as rather than a full regex it is fixing most of the string, which could change at any time. I would suggest doing something similar to 
```
if "Error" in pod_log and "exists from multiple origins" in pod_log:
      # Error detected
``` 
If you are only interested about the specific failure, otherwise using only [Ee]rror would be even better.

Reply:
I did think about that, but then those two strings could be in different lines and cause false positive. This way I can ensure they appear in the same line.